packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -84,55 +98,62 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
       scale = 1 / scale;
     }
 
-    return Scaffold(
-      body: ScannerOverlay(
-        restartCamera: _resumeImageStream,
-        stopCamera: _stopImageStream,
-        model: _model,
-        scannerWidget: Transform.scale(
-          scale: scale,
-          child: Center(
-            child: CameraPreview(
-              _controller!,
-            ),
-          ),
+    return Transform.scale(
+      scale: scale,
+      child: Center(
+        key: ValueKey<bool>(stoppingCamera),
+        child: CameraPreview(
+          _controller!,
         ),
       ),
     );
   }
 
   Future<void> _startLiveFeed() async {
+    stoppingCamera = false;
     final CameraDescription camera = cameras[_cameraIndex];
-    _controller = CameraController(
+
+    final CameraController cameraController = CameraController(
       camera,
       ResolutionPreset.high,
       enableAudio: false,
     );
-    _controller!.setFocusMode(FocusMode.auto);
-    _controller!.lockCaptureOrientation(DeviceOrientation.portraitUp);
+    cameraController.setFocusMode(FocusMode.auto);
+    cameraController.lockCaptureOrientation(DeviceOrientation.portraitUp);
 
-    _controller!.initialize().then((_) {
-      if (!mounted) {
-        return;
+    _controller = cameraController;
+
+    // If the controller is updated then update the UI.
+    cameraController.addListener(() {
```

Comment:
we keep adding this listener every time `_startLiveFeed` is called, shouldn't we do it just once?

Reply:
We only call `_startLiveFeed` once (and again but then the controller is already disposed) so it shouldn't matter.

Reply:
we also call `_startLiveFeed` from `LifeCycleManager`.`onResume`, so in the two cases you described below.